packages/customWidgets/calendar-web/src/components/CalendarContainer.ts

```diff
@@ -235,13 +245,33 @@ export default class CalendarContainer extends Component<Container.CalendarConta
         this.subscriptionContextHandles = [];
 
         if (mxObject) {
+            const attributePathValues = this.props.startDateAttribute.split("/");
+
+            if (attributePathValues.length > 2) {
+                const [referenceAttribute, , attr] = attributePathValues;
+
+                this.subscriptionContextHandles.push(
+                    window.mx.data.subscribe({
+                        guid: mxObject.get(referenceAttribute) as string,
+                        attr,
+                        callback: () => this.loadEvents(mxObject)
+                    })
+                );
+            }
             this.subscriptionContextHandles.push(
                 window.mx.data.subscribe({
                     guid: mxObject.getGuid(),
-                    attr: this.props.startDateAttribute,
-                    callback: () => this.loadEvents(mxObject)
+                    attr: attributePathValues[0],
+                    callback: () => {
+                        this.loadEvents(mxObject);
+
+                        if (attributePathValues.length > 2) {
+                            this.resetSubscriptions(mxObject);
+                        }
```

Comment:
Not entirely sure how this works and whether it's possible, but is it not possible to add this reset to the callback in the previous `window.mx.data.subscribe` (L257)?

Reply:
No, because the callback in L257 is only triggered when the associated object attribute value changes. This callback is triggered when the associated object changes. In this situation, the calendar needs to resubscribe to the attribute of the newly associated object.